playbooks/adhoc/zabbix_setup/clean_zabbix.yml

```diff
@@ -2,67 +2,51 @@
 - hosts: localhost
   gather_facts: no
   vars:
-    # Use this for local ZAIO
     g_zserver: http://localhost/zabbix/api_jsonrpc.php
-
     g_zuser: Admin
     g_zpassword: zabbix
   roles:
-  - ../roles/os_zabbix
+  - ../../../roles/os_zabbix
   post_tasks:
 
-  - zbxapi:
+  - zbx_template:
       server: "{{ g_zserver }}"
       user: "{{ g_zuser }}"
       password: "{{ g_zpassword }}"
-      zbx_class: Template
       state: list
-      params:
-        output: extend
-        search:
-          host: 'Template Heartbeat'
+      name: 'Template Heartbeat'
     register: templ_heartbeat
 
-  - zbxapi:
+  - zbx_template:
       server: "{{ g_zserver }}"
       user: "{{ g_zuser }}"
       password: "{{ g_zpassword }}"
-      zbx_class: Template
       state: list
-      params:
-        output: extend
-        search:
-          host: 'Template App Zabbix Server'
+      name: 'Template App Zabbix Server'
     register: templ_zabbix_server
 
-  - zbxapi:
+  - zbx_template:
       server: "{{ g_zserver }}"
       user: "{{ g_zuser }}"
       password: "{{ g_zpassword }}"
-      zbx_class: Template
       state: list
-      params:
-        output: extend
-        search:
-          host: 'Template App Zabbix Agent'
+      name: 'Template App Zabbix Agent'
     register: templ_zabbix_agent
 
-  - zbxapi:
+  - zbx_template:
       server: "{{ g_zserver }}"
       user: "{{ g_zuser }}"
       password: "{{ g_zpassword }}"
-      zbx_class: Template
       state: list
     register: templates
 
   - debug: var=templ_heartbeat.results
 
-  - zbxapi:
+  - zbx_template:
       server: "{{ g_zserver }}"
       user: "{{ g_zuser }}"
       password: "{{ g_zpassword }}"
-      zbx_class: Template
       state: absent
-      params: "{{templates.results | difference(templ_zabbix_agent.results) | difference(templ_zabbix_server.results) | oo_collect('templateid') }}"
+    with_items: "{{templates.results | difference(templ_zabbix_agent.results) | difference(templ_zabbix_server.results) | oo_collect('host') }}"
     register: template_results
```

Comment:
Consider removing this since you never use "template_results" anywhere (or actually use it)